cdragontoolbox/tftdata.py

```diff
@@ -180,15 +255,20 @@ def parse_traits(self, map22):
 
             effects = []
             for trait_set in trait.getv("mTraitSets"):
-                variables = []
+                variables = {}
+                increment = 0
                 for effect in trait_set.getv("effectAmounts", []):
-                    variables.append({"name": effect.getv("name"), "value": effect.getv("value")})
+                    name = effect.getv("name").s if effect.__contains__("name") else "unknown" + str(increment)
+                    name = "unknown" + str(increment) if name == "null" else "unknown" + str(increment)
```

Comment:
The ternary returns the same value in both cases.

Moreover, you should probably use `None` instead of `"null"` as "nothing" value.